### DISCUSSION TITLE
PWA/"Mobile App" Support

### DISCUSSION THREAD

**Discussion by nkringle** [mealie-recipes/mealie#929] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Basically I want to be able to have my phone/computer treat mealie like any other app. I created a bookmark on my iPhone which sort of accomplishes this, but with some limitations (there are some layout issues with resolution and mainly the lack of a "back" function - since the homepage bookmark doesn't show the normal safari back button, etc.) 

**(Optional): Suggest A Solution**
It seems like a [PWA](https://web.dev/progressive-web-apps/) may be a good solution here. I know there are a bunch of libraries and [getting started guides](https://www.vuemastery.com/blog/getting-started-with-pwas-and-vue3/) that tie into vue to do this, but I am not personally familiar with implementing them so I won't pretend to know too much 😁. 

I'm not really asking for a full native mobile app because I am sure you probably don't want to take that on and I don't know if that would add much benefit. :) 

**Additional Information**
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature
 -> Maybe, I haven't done any vue development so I'm a little hesitant to go this on my own right now.

**Comment by hay-kot** [comment]
As far as I understand it there is already PWA support for Mealie. Are you saying it's missing features or that it isn't working at all? 

I just checked and I have a PWA setup on my IPad and Android device. 

> I'm not really asking for a full native mobile app because I am sure you probably don't want to take that on and I don't know if that would add much benefit. :)

If I can get 1-2 other core developers I'm 100% down to build a Flutter App for Mealie after v1 get's release. So who knows!

**Reply by nkringle** [reply]
If there is PWA support it isn't coming up quite how I'd expect. It definitely is 80 or 90% there. I think there could be some improvements though. Namely, if you go to search for a recipe and click into something, there isn't a way (at least on my phone) to go back to my search. 


Additionally, the bottom menu is a little funky using an iPhone 13 Pro. (This screenshot does not do a good job of showing the rounded corners but you sort of get the idea that the icons are in the way of the bar that takes you home/edge of the screen. Also the search back arrow is back to my iphone's app search, not mealie search.

![image](https://user-images.githubusercontent.com/11429076/149578423-9863be20-3933-4e24-a4e7-c666c1b5b3b0.png)

Maybe it would be helpful to add a prompt of some kind to install the PWA? I'm fairly technical but most people I know (and would want to share my mealie instance with) are not.

**Reply by nkringle** [reply]
A full app would be pretty cool if I could give it the url of mealie and have it just work. 😁

**Reply by hay-kot** [reply]
>  Namely, if you go to search for a recipe and click into something, there isn't a way (at least on my phone) to go back to my search.

Not PWA related. Just a poor user experience, Could you create an issue for this? I think it may warrant a change to the flow.

> Additionally, the bottom menu is a little funky using an iPhone 13 Pro. (This screenshot does not do a good job of showing the rounded corners but you sort of get the idea that the icons are in the way of the bar that takes you home/edge of the screen. Also the search back arrow is back to my iphone's app search, not mealie search.

This will go away with v1. I had to do-away with the bottom bar because of the new IPhone update :(

> Maybe it would be helpful to add a prompt of some kind to install the PWA? 

I think this may be a browser thing and they only prompt you if you've visited the site so many times. However there are some articles that suggest it's possible to tie into it. https://web.dev/customize-install/ - at any rate it needs some further investigation. 

It also may be the Safari (therefore all iOS devices) don't support the PWA banner anyways. https://stackoverflow.com/questions/51160348/pwa-how-to-programmatically-trigger-add-to-homescreen-on-ios-safari

**Reply by nkringle** [reply]
> Not PWA related. Just a poor user experience, Could you create an issue for this? I think it may warrant a change to the flow.

Sure!

> It also may be the Safari (therefore all iOS devices) don't support the PWA banner anyways.

I just tried it in chrome and did not get any prompts (and I never use chrome on my phone). 🤷🏻‍♂️

**Reply by nkringle** [reply]
https://github.com/hay-kot/mealie/issues/931 for the UX/back feature

**Reply by hay-kot** [reply]
> I just tried it in chrome and did not get any prompts (and I never use chrome on my phone).> 

All iOS browsers use Safari under the hood so if you're using it on your IPhone you're still bound to Safari's rules/limitations.

For example, I get a prompt right away on Android.

<img height="600"  src="https://user-images.githubusercontent.com/64056131/149584316-dcdb1ad7-c017-4a8c-8731-ebc1d98252c6.jpg"><img>


From what I understand this is because of Apple's strong payment terms in the app store that wouldn't apply to a PWA, so they don't go out of their way to make PWAs a first class option for iOS users. So it's likely this will never change.

**Reply by nkringle** [reply]
I didn't know that! Maybe another vote for a native app then!
